cpp/src/arrow/util/value_parsing.h

```diff
@@ -163,6 +166,20 @@ struct StringConverter<DoubleType> {
   const char decimal_point;
 };
 
+template <>
+struct StringConverter<HalfFloatType> {
```

Comment:
Should have added some tests for value parsing and formatting. @kou

Reply:
Ah, I thought tests added to `scalar_cast_test.cc` cover this.
I missed that we have `value_parsing_test.cc`.

@ClifHouck Could you add tests for this to `value_parsing_test.cc` as a follow-up task?

Reply:
And also `formatting_util_test.cc`

Reply:
I'll take a look and add tests.